front/pages/poke/[wId]/data_sources/[name]/index.tsx

```diff
@@ -45,6 +46,7 @@ export const getServerSideProps = withSuperUserAuthRequirements<{
     googleDrivePdfEnabled: boolean;
     googleDriveLargeFilesEnabled: boolean;
     githubCodeSyncEnabled: boolean;
+    whiteListedChannelPatterns: string | string[];
```

Comment:
On the FE side I'd suggest to either always deal with string or always an array but let's avoid those types that requires conditional code down the line.